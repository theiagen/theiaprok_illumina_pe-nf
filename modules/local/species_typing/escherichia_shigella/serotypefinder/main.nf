process SEROTYPEFINDER {
    tag "$meta.id"
    label 'process_low'
    
    container "us-docker.pkg.dev/general-theiagen/staphb/serotypefinder:2.0.1"

    input:
    tuple val(meta), path(assembly)
    
    output:
    tuple val(meta), path ("*_results_tab.tsv"), emit: serotypefinder_report
    tuple val(meta), path ("STF_SEROTYPE"), emit: serotypefinder_serotype
    path "versions.yml", emit: versions
    
    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    # capture date and version
    date | tee DATE

    # Check if assembly is gzipped and decompress if needed
    if [[ "${assembly}" == *.gz ]]; then
        echo "Input assembly is gzipped. Decompressing..."
        decompressed_assembly=\$(basename "${assembly}" .gz)
        gunzip -c "${assembly}" > "\${decompressed_assembly}"
        assembly="\${decompressed_assembly}"
        echo "Decompressed to \${decompressed_assembly}"
        
        # Run serotypefinder on the decompressed assembly
        serotypefinder.py -i \${assembly}  -x -o .
    else
        echo "Input assembly is not gzipped. Proceeding with original file."
        serotypefinder.py -i ${assembly}  -x -o .
    fi

    mv results_tab.tsv ${prefix}_results_tab.tsv
    
    parse_serotypefinder.py --input ${prefix}_results_tab.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python3 --version | cut -d' ' -f2)
        serotypefinder: 2.0.2
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "na" > ${prefix}_results_tab.tsv
    echo "na" > STF_SEROTYPE

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python3 --version | cut -d' ' -f2)
        serotypefinder: 2.0.2
    END_VERSIONS
    """
}