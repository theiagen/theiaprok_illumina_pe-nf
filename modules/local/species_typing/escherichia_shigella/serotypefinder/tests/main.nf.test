nextflow_process {
    name "Test Process SEROTYPEFINDER"
    script "../main.nf"
    process "SEROTYPEFINDER"
    
    test("serotypefinder test") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    file("$projectDir/bin/473675_contigs.fasta", checkIfExists: true)
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.serotypefinder_report,
                    process.out.serotypefinder_serotype,
                    process.out.versions
                    ).match() }
            )
        }
    }
    
    test("serotypefinder - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    file("$projectDir/bin/473675_contigs.fasta", checkIfExists: true)
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
