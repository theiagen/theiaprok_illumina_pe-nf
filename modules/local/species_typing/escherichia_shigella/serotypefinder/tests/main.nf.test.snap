{
    "serotypefinder - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_results_tab.tsv:md5,51bb9b62d9be48ac08dfb0ec931632e2"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "STF_SEROTYPE:md5,51bb9b62d9be48ac08dfb0ec931632e2"
                    ]
                ],
                "2": [
                    "versions.yml:md5,db1a709a777e4e407f851a431403cf55"
                ],
                "serotypefinder_report": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample_results_tab.tsv:md5,51bb9b62d9be48ac08dfb0ec931632e2"
                    ]
                ],
                "serotypefinder_serotype": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "STF_SEROTYPE:md5,51bb9b62d9be48ac08dfb0ec931632e2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,db1a709a777e4e407f851a431403cf55"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.8"
        },
        "timestamp": "2025-06-24T19:57:54.160586468"
    },
    "serotypefinder test": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample_results_tab.tsv:md5,4e36723ab36b61aaf9ca7274576f8e70"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "STF_SEROTYPE:md5,79fc5cdb8d552909e3ee918ba71f0942"
                ]
            ],
            [
                "versions.yml:md5,db1a709a777e4e407f851a431403cf55"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.8"
        },
        "timestamp": "2025-06-24T19:57:49.683564849"
    }
}